Content.Server/Fluids/EntitySystems/AbsorbentSystem.cs

```diff
@@ -185,13 +198,39 @@ private bool TryPuddleInteract(EntityUid user, EntityUid used, EntityUid target,
             return true;
         }
 
-        var transferMax = absorber.PickupAmount;
-        var transferAmount = available > transferMax ? transferMax : available;
+        // hack to compensate loss due to fractions that would be periodical
+        // and make sure (available / Ratio) * Ratio is not less than available
+        // otherwise the absorber would very annoyingly never fill
+        // it's not necessary if the ratio is not periodical in a decimal base
+        if (PuddleSystem.EvaporationReagentRatio % 2 != 0 &&
+                PuddleSystem.EvaporationReagentRatio % 5 != 0){
+            available += FixedPoint2.Epsilon;
+        }
+
+        var dirtVolumeAvailable = puddleSolution.Volume - puddleSolution.GetReagentQuantity(PuddleSystem.EvaporationReagent);
+
+        var transferAmount = FixedPoint2.Min(dirtVolumeAvailable,
+                                             available * PuddleSystem.EvaporationReagentRatio,
+                                             absorber.PickupAmount);
+
+        var prospectWaterTransfer = FixedPoint2.Min(transferAmount / PuddleSystem.EvaporationReagentRatio,
+                                                    absorberSoln.GetReagentQuantity(PuddleSystem.EvaporationReagent));
+
+        // This var exists for no other purpose than code readability
+        var prospectAvailableVolume = absorberSoln.AvailableVolume + prospectWaterTransfer;
```

Comment:
This kinda contrived solution is what allowed me to remove the extra split while still not losing any fluid and also adding a few asserts in the way.
This also allowed me to find a possible bug where I was assuming the amount I asked to transfer was being really transfer without checking the return of RemoveReagent.